service-registry/src/main/java/io/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java

```diff
@@ -95,6 +98,36 @@ public void init() {
     eventBus.register(this);
   }
 
+  protected void initAppManager() throws InstantiationException, IllegalAccessException, ClassNotFoundException {
+    appManager = new AppManager(eventBus);
+
+    String microserviceVersionFactoryClass = serviceRegistryConfig.getMicroserviceVersionFactory();
+    if (microserviceVersionFactoryClass == null) {
+      return;
```

Comment:
It's better to throw exception here.

Reply:
in this version, we did not remove old cacheManager, and new cacheManager only used by edge.
so microserviceVersionFactoryClass is null, means not work with edge, will use old cacheManager, can not throw exception

Reply:
Can we add a comment to the code for it?

Reply:
done.